pkg/services/ngalert/notifier/multiorg_alertmanager_test.go

```diff
@@ -144,6 +150,93 @@ func TestMultiOrgAlertmanager_AlertmanagerFor(t *testing.T) {
 	}
 }
 
+func TestMultiOrgAlertmanager_addOrUpdateConfiguration(t *testing.T) {
+	kvStore := newFakeKVStore(t)
+	reg := prometheus.NewPedanticRegistry()
+	m := metrics.NewNGAlert(reg)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	configStore := &FakeConfigStore{
+		configs: map[int64]*models.AlertConfiguration{},
+	}
+
+	cfg := &setting.Cfg{DataPath: t.TempDir()}
+
+	mam, initErr := NewMultiOrgAlertmanager(cfg, configStore, nil, kvStore, m.GetMultiOrgAlertmanagerMetrics(), log.New("testlogger"))
+	require.NoError(t, initErr)
+
+	t.Run("Adds missing Alertmanager and applies existing configuration", func(t *testing.T) {
```

Comment:
Apologies if I'm missing something but aren't these tests testing the same as `TestMultiOrgAlertmanager_SyncAlertmanagersForOrgs`? The only difference here I can see is that you're providing explicit configuration as opposed to just using the default - which you can do by modifying the above?

Reply:
These tests are more focused on the specific logic of what happens when a combination of existing configuration and db configuration changes. The tests above do not cover all possible combinations (for example when there is no config in db for existing organization) and they test many things at once - getting data from orgs table, getting data from config tables, iterating orgs, removing missing etc. I find such tests are pretty hard to read.